src/extrapolation/extrap_prep.jl

```diff
@@ -0,0 +1,138 @@
+"""
```

Comment:
I had an implementation that didn't use dispatch so much but rather had a bunch of (read: ton of) compile-time if-clauses. Would that be easier to understand, do you think?